### DISCUSSION TITLE
Images lazyload (tables)

### DISCUSSION THREAD

**Discussion by d8vjork** [filamentphp/filament#6560] [Ideas]
As images might be a heavy thing, I think adding the possibility to not load everything at once but using some img lazyloading JS library could be a very nice thing to have.

And maybe adding a `Tables\Columns\ImageColumn::make(...)->lazyLoad()` method to let users decide when to enable or not?

**Comment by danharrin** [comment]
I think lazy images are built into browsers nowadays, so we wouldnt need a JS library

**Reply by d8vjork** [reply]
Really? Didn't notice... Anyway my table page hangs whenever I enter the resource (this one having multiple QR codes as images showing in the table), with a 10 row table is already suffering to load at the beginning (which shouldn't)

**Reply by danharrin** [reply]
`ImageColumn::make()->extraImgAttributes(['loading' => 'lazy'])` will lazy load all images off screen. Also be sure to optimise your images so they aren't full-size in the table.

**Reply by d8vjork** [reply]
@danharrin Cooool! Exactly what I needed (didn't know of that attribute). Thanks
